addons/html_editor/static/src/main/hint_plugin.js

```diff
@@ -70,36 +70,33 @@ export class HintPlugin extends Plugin {
     /**
      * @param {HTMLElement} [root]
      */
-    updateHints(root = this.editable) {
-        this.clearHints(root);
-        this.makeEmptyBlockHints(root);
-        this.updateTempHint(this.shared.getSelectionData());
-    }
-
-    makeEmptyBlockHints(root) {
-        for (const { selector, hint } of this.resources.emptyBlockHints) {
-            for (const el of selectElements(root, selector)) {
-                // @todo: consider using isEmptyBlock instead.
-                if (isEmpty(el) && !isProtected(el) && !isProtecting(el)) {
-                    this.makeHint(el, hint);
-                }
-            }
+    updateHints() {
+        const selectionData = this.shared.getSelectionData();
+        const editableSelection = selectionData.editableSelection;
+        if (this.hint) {
+            const blockEl = closestBlock(editableSelection.anchorNode);
+            this.removeHint(this.hint);
+            this.removeHint(blockEl);
```

Comment:
We will now only have a single hint in editor so `this.removeHint(this.hint)` shall be enough.

Reply:
@deso-odoo When using Powerbox to convert a block element to `text`, `this.hint` points to the element that is being transformed into a `<p>` and removes the hint from it. However, the newly created `<p>` still keeps the hint from the original block element, such as `Heading 1`. This prevents the hint from updating to `Type '/' for commands` due to a condition that prevents hint changes if one is already present on the `<p>`. Therefore, it's necessary to also remove the hint from the `anchorElement`.